schema/ksl-schema.kdl

```diff
@@ -0,0 +1,617 @@
+                                    * node properties MUST by replaced by key.
+                                    * node arguments MUST be replaced by order of appearance.
+                                    * `prop` definitions MUST be replaced by key (their first argument)
+                                    * `arg` definitions MUST be replaced based on _order of
+                                       appearance_. That is, the first `arg` in @ksl:ref `B` till be
+                                       merged into the first `arg` in preceding @ksl:ref `A`.
+                                    * For all other components:
+                                        * If the definition specified is marked as
+                                          `repeatable`, then all definitions using that node
+                                          will be concatenated, with later `ref`s
+                                          concatenating definitions after the previous `ref`'s
+                                          definitions.
+                                        * If the definition is NOT marked as `repeatable`,
+                                          it will be replaced by subsequent `ref`s.
+                                    
+                                    Once all `ref` children are resolved, the containing
+                                    node's own items will override anything defined by
+                                    `ref`s, using the same rules as above (essentially, the
+                                    current node is treated as a 'final `ref`').
+                                    
+                                    If both an ID argument and a `path` are provided,
+                                    the ID will take precedence and, if not found, fall
+                                    back to the path. For `id` and `path` children,
+                                    precedence is in order of appearance, regardless of
+                                    whether the child is an `id` or a `path`. 
+                                    
+                                    If no items resolve into a valid @ksl:reference,
+                                    validation MUST error, unless the @ksl:ref is configured
+                                    as `optional`, in which case validation MAY warn,
+                                    but MUST NOT fail.
+                                    """ {
+                                    repeatable
+                                    arg title="ID @ksl:reference to another node" {
+                                        type string
+                                        optional
+                                    }
+                                    prop path title="KPath @ksl:reference to another node." {
+                                        type string
+                                        format kpath
```

Comment:
I assume this means we have to finalize the KPath 2.0 spec before we can finalize this spec?

Reply:
yep. And possibly make a small modification to the v2 spec itself to add `kpath` as a "reserved" string format.

Reply:
Do you mean as a type annotation? Like these? https://github.com/kdl-org/kdl/blob/main/draft-marchan-kdl2.md#reserved-type-annotations-for-strings

Something like `(kpath)"<whatever>`?

Reply:
the formats correspond to type annotations, but they do not specify the annotation that needs to be present. That is, `format kpath` will validate `"foo"` as if it were `(kpath)"foo"`, but the annotation is not required. To do that, use the `annotation` node.